libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -213,6 +259,23 @@ bool HTTPClient::begin(String host, uint16_t port, String uri, String httpsFinge
     return true;
 }
 
+bool HTTPClient::begin(String host, uint16_t port, String uri, const uint8_t httpsFingerprint[20])
```

Comment:
Code duplication is increasing with all these begin() methods. But let's fix that later as a cleanup step.

Reply:
One suggestion voiced elsewhere was to change the way HTTPClient works to be similar to PubSubClient. That is, HTTPClient receives a prepared Client object (which can be WiFiClient or WiFiClientSecure or any other client). This object already has fingerprint set if necessary. If you think this is a good idea, i can prepare a PR.

Reply:
@igrr That's a good idea.  Maybe for this immediate round we can shelve it and then migrate to that way afterwards?  I'm already at 71 updated files here, and I'd rather not boil any more of the ocean in this pot than necessary.

We'd need to change `axTLS::WiFiClientSecure` to allow this `setFingerprint()` call and, when used, do an immediate check on `connect()`.  Then only one TLSClient class in HTTPClient would be needed  and we can decorate the older ones as __deprecated__.

Reply:
Sure sure, no need to do anything in this PR. Once this is merged, I will prepare a PR about HTTPClient.